webapp-django/crashstats/crashstats/models.py

```diff
@@ -927,50 +921,27 @@ class SignatureFirstDate(SocorroMiddleware):
         )
     }
 
-    def get_dates(self, signatures):
-        """A highly optimized version, that returns a dictionary where
-        the keys are the signature and the values are dicts that look
-        like this for example::
+    def get(self, *args, **kwargs):
+        params = self.parse_parameters(kwargs)
 
-            {
-                'first_build': u'20101214170338',
-                'first_date': datetime.datetime(
-                    2011, 1, 17, 21, 24, 18, 979850, tzinfo=...)
-                )
-            }
+        if 'signatures' not in params:
+            raise MissingArgumentError('signatures')
 
-        """
-        dates = {}
-        missing = {}
-        for signature in signatures:
-            # calculate a good cache key for each signature
-            cache_key = 'signature_first_date-{}'.format(
-                hashlib.md5(signature.encode('utf-8')).hexdigest()
-            )
-            cached = cache.get(cache_key)
-            if cached is not None:
-                dates[signature] = cached
-            else:
-                # remember the cache keys of those we need to look up
-                missing[signature] = cache_key
+        hits = list(
+            Signature.objects
+            .filter(signature__in=params['signatures'])
+            .values('signature', 'first_build', 'first_date')
+        )
 
-        if missing:
-            hits = super(SignatureFirstDate, self).get(
-                signatures=missing.keys()
-            )['hits']
-
-            for hit in hits:
-                signature = hit.pop('signature')
-                cache.set(
-                    # get the cache key back
-                    missing[signature],
-                    # what's left when 'signature' is popped
-                    hit,
-                    # make it a really large number
-                    60 * 60 * 24
-                )
-                dates[signature] = hit
-        return dates
+        # NOTE(willkg): The original implementation return first_build as a
+        # string, so we convert it here.
+        for hit in hits:
+            hit['first_build'] = str(hit['first_build'])
+
+        return {
+            'hits': hits,
+            'total': len(hits)
+        }
```

Comment:
This didn't have any tests before. I should add one to at least verify the shape of the results.